### PR TITLE
Add support to nomdex to query against multiple indexes.

### PR DESCRIPTION
**Pull request by willhite**

### CODE REVIEW

File: samples/go/nomdex/expr.go [hunk 1]
```diff
@@ -16,22 +16,43 @@ import (
 type expr interface {
 	ranges() queryRangeSlice
 	dbgPrintTree(w io.Writer, level int)
+	indexName() string
+	Iterator(im *indexManager) types.SetIterator
```
Comment (aboodman) [inline samples/go/nomdex/expr.go]: Did you mean for this to be public?

File: samples/go/nomdex/expr.go [hunk 2]
```diff
@@ -16,22 +16,43 @@ import (
...
 }
 
 // logExpr represents a logical 'and' or 'or' expression between two other expressions. e.g.
```
Comment (aboodman) [inline samples/go/nomdex/expr.go]: e.g. what?

File: samples/go/nomdex/expr.go [hunk 3]
```diff
@@ -16,22 +16,43 @@ import (
...
+	return le.idxName
+}
+
+func (le logExpr) Iterator(im *indexManager) types.SetIterator {
+	if le.idxName != "" {
+		return unionizeIters(iteratorsFromRanges(im.indexes[le.idxName], le.ranges()))
+	}
+
+	var iter types.SetIterator
+	switch le.op {
```
Comment (aboodman) [inline samples/go/nomdex/expr.go]: 'le op' heh.

File: samples/go/nomdex/parser.go [hunk 4]
```diff
@@ -12,22 +12,31 @@ import (
...
-  expr := expr boolop relExpr | group
-  relExpr := index relOp value
-  group := '(' expr ')' | relExpr
+  expr := expr boolOp compExpr | group
+  compExpr := indexToken compOp value
+  group := '(' expr ')' | compExpr
   boolOp := 'and' | 'or'
-  relOp := '=' | '<' | '<=' | '>' | '>='
-  value := "<string>" | int | float
+  compOp := '=' | '<' | '<=' | '>' | '>='
```
Comment (aboodman) [inline samples/go/nomdex/parser.go]: did you consider adding != ? I ran into this when testing.

**Comment by aboodman** [comment]
Regarding the functionality, LGTM. Regarding the code, maybe ben would be better since he reviewed the set iterator stuff?

**Comment by aboodman** [comment]
Nevermind, I will do.

File: samples/go/nomdex/nomdex_find.go [hunk 5]
```diff
@@ -26,28 +26,31 @@ var find = &util.Command{
...
+	flagSet.StringVar(&dbPath, "db", "", "database containing index")
 	outputpager.RegisterOutputpagerFlags(flagSet)
 	return flagSet
 }
 
 func runFind(args []string) int {
 	query := args[0]
-	if indexPath == "" {
+	if dbPath == "" {
 		fmt.Fprintf(os.Stderr, "Missing required 'index' arg\n")
```
Comment (aboodman) [inline samples/go/nomdex/nomdex_find.go]: error message now out of date

**Comment by aboodman** [comment]
lgtm w/ nits

**Comment by willhite** [comment]
@aboodman These two commits address the issues you found today with nomdex. If it still looks good to you, I'll land these changes and mark the nomdex issue #2405 and #2402 as fixed. And if you agree I'll mark #2114 (the main nomdex bug) as complete.

I'll add the help text for 'nomdex find' and possibly implement the '!=' operator in a later patch. The != operator would require a change to the range logic but shouldn't be too difficult.

**Comment by aboodman** [comment]
Agreed to all

**Pull request merged**